app/services/green_lanes/determine_category.rb

```diff
@@ -25,6 +25,14 @@ def call
       end
     end
 
```

Comment:
These methods are repeating same logic over and over and contradicting DRY prinsiple. Instead you can write two parameterised methods, something like
`def category_assessments(category)
    goods_nomenclature.applicable_category_assessments.select { |ca| ca.theme.category == category }
  end

  def has_exemptions(category)
    category_assessments(category).select {|ca| ca.exemptions.any?}
  end`
  
  These two methods are enough to detemine the result in categories emthod

Reply:
Good point, 
I removed the duplication without undermining readability.
It should be good.